test/e2e_env/multizone/localityawarelb/locality_aware_multizone_hybrid_egress.go

```diff
@@ -0,0 +1,134 @@
+	const mesh = "locality-aware-lb-egress"
+	const namespace = "locality-aware-lb-egress"
+
+	meshLoadBalancingStrategyDemoClient := fmt.Sprintf(`
+type: MeshLoadBalancingStrategy
+name: mlbs-1
+mesh: %s
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef:
+        kind: MeshService
+        name: test-server_locality-aware-lb-egress_svc_80
+      default:
+        localityAwareness:
+          crossZone:
+            failover:
+              - from: 
+                  zones: ["kuma-4"]
+                to:
+                  type: Only
+                  zones: ["kuma-5"]
+              - from: 
+                  zones: ["kuma-4"]
+                to:
+                  type: Only
+                  zones: ["kuma-1-zone"]
+              - from: 
+                  zones: ["kuma-4"]
+                to:
+                  type: Any`, mesh)
+
+	BeforeAll(func() {
+		// Global
+		Expect(NewClusterSetup().
+			Install(ResourceUniversal(samples.MeshMTLSBuilder().WithName(mesh).WithEgressRoutingEnabled().Build())).
+			Install(YamlUniversal(meshLoadBalancingStrategyDemoClient)).
+			Setup(multizone.Global)).To(Succeed())
+		Expect(WaitForMesh(mesh, multizone.Zones())).To(Succeed())
+
+		// Universal Zone 4
+		Expect(NewClusterSetup().
+			Install(DemoClientUniversal(
+				"demo-client_locality-aware-lb_svc",
+				mesh,
+				WithTransparentProxy(true),
+			)).
+			Install(TestServerUniversal("test-server-zone-4", mesh,
+				WithServiceName("test-server_locality-aware-lb-egress_svc_80"),
+				WithArgs([]string{"echo", "--instance", "test-server-zone-4"}),
+			)).
+			Setup(multizone.UniZone1),
+		).To(Succeed())
+
+		// Universal Zone 5
+		Expect(NewClusterSetup().
+			Install(DemoClientUniversal(
+				"demo-client_locality-aware-lb_svc",
+				mesh,
+				WithTransparentProxy(true),
+			)).
+			Install(TestServerUniversal("test-server-zone-5", mesh,
+				WithServiceName("test-server_locality-aware-lb-egress_svc_80"),
+				WithArgs([]string{"echo", "--instance", "test-server-zone-5"}),
+			)).
+			Setup(multizone.UniZone2),
+		).To(Succeed())
+
+		// Kubernetes Zone 1
+		Expect(NewClusterSetup().
+			Install(NamespaceWithSidecarInjection(namespace)).
+			Install(democlient.Install(democlient.WithMesh(mesh), democlient.WithNamespace(namespace))).
+			Install(testserver.Install(
+				testserver.WithName("test-server"),
+				testserver.WithMesh(mesh),
+				testserver.WithNamespace(namespace),
+				testserver.WithEchoArgs("echo", "--instance", "test-server-zone-1"),
+			)).
+			Setup(multizone.KubeZone1)).ToNot(HaveOccurred())
+	})
+
+	E2EAfterAll(func() {
+		Expect(multizone.UniZone1.DeleteMeshApps(mesh)).To(Succeed())
+		Expect(multizone.UniZone2.DeleteMeshApps(mesh)).To(Succeed())
+		Expect(multizone.KubeZone1.TriggerDeleteNamespace(namespace)).To(Succeed())
+		Expect(multizone.Global.DeleteMesh(mesh)).To(Succeed())
+	})
+
+	It("should route based on defined strategy with egress enabled", func() {
+		Eventually(func() (map[string]int, error) {
```

Comment:
We can maybe make it similar to the current test without egress.
1. Check traffic without policy
2. Apply policy
3. Check when app is down
4. Check when the next is down

and we can add `E2EAfterEach` which removes the policy

Reply:
changed test, found bug. And changed test again to cover more complicated use case:

```
1. Check traffic without policy
2. Apply policy
3. Check when app is down
4. Change policy priorities
5. Check traffic
6. Check when the next is down
```